op-node/rollup/derive/batches.go

```
@@ -133,6 +133,12 @@ func checkSingularBatch(cfg *rollup.Config, log log.Logger, l1Blocks []eth.L1Blo
 		return BatchDrop
 	}
 
+	// Future forks that contain upgrade transactions must be added here.
+	if (cfg.IsInteropActivationBlock(batch.Timestamp)) && len(batch.Transactions) > 0 {
+		log.Warn("dropping batch with user transactions in fork activation block")
+		return BatchDrop
+	}
```

Do we have test coverage to show what would happen here? I think it would be an safe head stall until the sequencer reorged the unsafe chain to one that respects the "no user transactions" rule and rolled that up to L1.

Also: do we have checks that would catch a violation of this rule in the unsafe chain? e.g. in the execution engine we could return an error if `noTxPool` is not false for the activation block.

---

Will follow up with that. For now would like to get the upgrade txs in for the rehearsal rc betanet upgrade.

---

I've written unit and e2e tests for this rule when we wanted to add it to Isthmus: https://github.com/ethereum-optimism/optimism/pull/14797
You could probably easily adapt these tests.

---

TODO in follow-up